pkg/file/io.go

```diff
@@ -107,3 +107,18 @@ func SplitWriteAll(ctx context.Context, s Splitter, r io.Reader, l int64, toEncr
 	}
 	return addr, nil
 }
+
+type Loader interface {
+	// Load a reference in byte slice representation and return all content associated with the reference.
+	Load(context.Context, []byte) ([]byte, error)
```

Comment:
I think since we are in the `bee` project, the `Load` and `Save` function should work directly with `swarm.Address`, instead of using `[]byte`.

This may require having some proxy implementation for manifest, where we have implementation with byte slice, which translates to `swarm.Address`.

Reply:
but then it means I need an extra abstraction that would use this one. I'm not sure why this is necessary. I only created this implementation to be compatible with the manta ray interfaces. I think that once other components need to use similar functionality we can generalize, but I'm not sure why to do so prematurely

Reply:
Yeah, I just do not think application should cater to library completely.